spec/requests/customers_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+RSpec.describe "CustomersControllers", type: :request do
+  describe "get customers_path" do
+    it "renders the index view" do
+      FactoryBot.create_list(:customer, 10)
+      get customers_path
+      expect(response).to render_template(:index)
+    end
+  end
+  describe "get customer_path" do
+    it "renders the :show template" do
+      customer = FactoryBot.create(:customer)
+      get customer_path(id: customer.id)
+      expect(response).to render_template(:show)
+    end
+    it "redirects to the index path if the customer id is invalid" do
+      get customer_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to customers_path
+    end
+  end
+  describe "get new_customer_path" do
+    it "renders the :new template" do
+      customer = FactoryBot.create(:customer)
+      get new_customer_path
+      expect(response).to render_template(:new)
+    end
+  end
+  describe "get edit_customer_path" do
+    it "renders the :edit template" do
+      customer = FactoryBot.create(:customer)
+      get edit_customer_path(id: customer.id)
+      expect(response).to render_template(:edit)
+    end
+  end
+  describe "post customers_path with valid data" do
+    it "saves a new entry and redirects to the show path for the entry" do
+      customer_attributes = FactoryBot.attributes_for(:customer)
```

Comment:
You can add alternative Factories by nesting them
```Ruby
# in factory file
factory :user do
  first_name { 'John' }

  factory :sample_user do
    first_name { Faker::Name.first_name }
  end
end
 
# in test
create :sample_user
```

You don't have to do this, but I thought it would be helpful for the future.
My example from this cheatsheet for Factorybot https://devhints.io/factory_bot